name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/TradeRepublicPDFExtractor.java

```diff
@@ -1037,7 +1040,124 @@ private void addDeliveryInOutBoundTransaction()
         addTaxesSectionsTransaction(pdfTransaction2, type);
         addFeesSectionsTransaction(pdfTransaction2, type);
     }
+    
+    private void addTaxInOutBoundTransaction()
+    {
+        DocumentType type = new DocumentType("(STEUERLICHER UMTAUSCH)", (context, lines) -> {
+            Pattern pDate = Pattern.compile("^(.*) DATUM (?<date>[\\d]{2}\\.[\\d]{2}\\.[\\d]{4})$");
+            context.put("transactionPosition", "0");
+
+            for (String line : lines)
+            {
+                Matcher mDate = pDate.matcher(line);
+                if (mDate.matches())
+                    context.put("date", mDate.group("date"));
+
+            }
+        });
+        this.addDocumentTyp(type);
+
+        Transaction<PortfolioTransaction> pdfTransaction = new Transaction<>();
+
+        Block firstRelevantLine = new Block("(^|^[\\d] )(Einbuchung"
+                        + "|Ausbuchung) .* "
+                        + "([\\.,\\d]+ Stk\\.)$");
+        type.addBlock(firstRelevantLine);
+        firstRelevantLine.set(pdfTransaction);
+
+        pdfTransaction //
+
+                .subject(() -> {
+                    PortfolioTransaction portfolioTransaction = new PortfolioTransaction();
+                    portfolioTransaction.setType(PortfolioTransaction.Type.DELIVERY_OUTBOUND);
+                    return portfolioTransaction;
+                })
+
+                // Is type --> "Einbuchung" change from DELIVERY_OUTBOUND to DELIVERY_INBOUND
+                .section("type")
+                .match("(^|^[\\d] )(?<type>"
+                                + "(Einbuchung|Ausbuchung)"
+                                + ") .* ([\\.,\\d]+ Stk\\.)$")
+                .assign((t, v) -> {
+                    if ("Einbuchung".equals(v.get("type")))
+                    {
+                        t.setType(PortfolioTransaction.Type.DELIVERY_INBOUND);
+                        t.setNote("Einstandskurs fehlt");
+                    }
+                })
```

Comment:
Missing translation "note" but i think... remove this line.
```suggestion
                .section("type")
                .match("^[\\d] (?<type>(Einbuchung|Ausbuchung)) .* [\\.,\\d]+ Stk\\.$")
                .assign((t, v) -> {
                    if ("Einbuchung".equals(v.get("type")))
                        t.setType(PortfolioTransaction.Type.DELIVERY_INBOUND);
                })
```

Reply:
Wenn im Dokument in Deutsch Ein/Ausbuchung steht wird der Nutzer auch "Einstandskurs fehlt" verstehen. Finde den Hinweis wichtig, da sich daraus ja die zu zahlenden Steuern bei späterer Veräußerung berechnen. Der Import jeglicher Einlieferungen mit Wert 0 führt ja zur Verfälschung des Einstandskurs (zum Beispiel bei Berichte-> Performance-> Wertpapiere in den Spalten Einstandskurs und Nicht realisierte Kurserfolge).

Finde den Hinweis hilfreich. Wenn irgendwann anderssprachige Trade Republic Dokumente auftauchen stimme ich dir zu - dann brauchen wir eine Übersetzung.

Reply:
Wenn der "Einstandskurs fehlt" sollten wir dann nicht mit "cancellation" oder "not supported" den Import markieren?
Wäre das nicht sinnvoller?

Reply:
Naja um den aktuellen Bestand abzubilden, ist das als Platzhalter ja schonmal nicht schlecht.
Die steuerlichen Details sehe ich persönlich eher optional.
Das Problem zieht sich durch sehr viele Importer, dass der Kurs bei Einlieferung hart auf 0 gesetzt wird. Das müsste man dann alles rausnehmen und Funktionsumfang reduzieren. Ich denke mit dem Einfügen einer Notiz ist der Gewinn größer.

L1059 kann ich noch anpassen. Ansonsten sind die doch gleich?